### ISSUE TITLE
Document how to delete data from the DLQ

### ISSUE THREAD

**Issue by jguay** [elastic/logstash#8795]
https://www.elastic.co/guide/en/logstash/current/dead-letter-queues.html
Can a section on how to clear data from DLQ be added?
If this involves restarting logstash (https://github.com/elastic/logstash/issues/8794) then it is a limitation which is important to know.

**Comment by yawboateng** [comment]
any news on this? my DLQ is full and causing constant restart of my containers

multiple lines of:
`[2018-03-15T14:02:39,728][ERROR][org.logstash.common.io.DeadLetterQueueWriter] cannot write event to DLQ: reached maxQueueSize of 1073741824`

**Comment by hakan1997** [comment]
Would really appreciate this as well

**Comment by zzt93** [comment]
Any update?

**Comment by excalq** [comment]
I just did `rm -rf /path/to/dlq/main/*` after inspecting the contents.

**Comment by gedge** [comment]
it seems, once logstash thinks that the DLQ has grown too big, it will never write to it again, no matter what you delete, because it never checks the disk

it only ever adds to its internal count of that size: https://github.com/elastic/logstash/blob/master/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java#L140

hence a restart is required 🤦‍♂

**Comment by ceeeekay** [comment]
It would be great to have the option to clear the DLQ, or have Logstash write over older failed events.

We have a LS input constantly reading the DLQ, moving the event into a single field as JSON, attaching the DLQ reason metadata, and sending it to a separate Elasticsearch 'dead-letter' index for inspection.

This is much preferred to having to attend to queues on a lot of separate LS nodes. 

Even in the event that there is a torrent of data arriving in the DLQ, it all arrives safely in ES, so the older content in the queue is not relevant to us at that point and can be safely deleted/written over.

**Comment by willemveerman** [comment]
I've got around this in kubernetes by configuring an output plugin in logstash which kills the container if the DLQ exceeds 80% of its pre-configured size.

The DLQ itself is at a path within the container, so when the pod is recycled it is cleared completely - as is logstash's internal DLQ count.

The sincedb files which track the log file reading position are in a dir bind-mounted from the host, so they survive the pod recycle

```
    input {
      dead_letter_queue {
        id => "kubernetes_dlq"
        path => "${PATH_DEAD_LETTER_QUEUE}"
        sincedb_path => {{ .Values.logstash.dead_letter_queue_sincedb_path | quote }}
        commit_offsets => true
        tags => ["dlq"]
      }
    }

    # If the DLQ is > 80% full kill the pod
    output {
      if "dlq" in [tags] {
        exec {
          command => "if (( $(du -sb {{ .Values.logstash.path_dead_letter_queue }}/main/ | cut -f1) > $(( {{ .Values.logstash.dead_letter_queue_max_bytes }}*80/100 )) )); then kill 1; fi &"
        }
      }
    }
```

**Comment by karenzone** [comment]
We've documented how to clear the DLQ, and added other DLQ information in this PR: https://github.com/elastic/logstash/pull/12959.

Here's a link to the doc: https://www.elastic.co/guide/en/logstash/current/dead-letter-queues.html#dlq-clear

**Comment by karenzone** [comment]
This issue to document how to delete data from the DLQ is resolved with #12959. 
The broader issue is still valid and being tracked in #10493.
